src/lib.rs

```diff
@@ -320,6 +320,224 @@ impl DltExtendedHeader {
+            0x4 => Some(Mtin::DltTraceMessage(MtinDltTraceMessage::DltTraceState)),
+            0x5 => Some(Mtin::DltTraceMessage(MtinDltTraceMessage::DltTraceVfb)),
+            //Reserved values
+            _ => None,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum MtinDltNetworkMessage {
+    ///Inter-Process-Communication.
+    DltNwTraceIpc,
+    ///CAN communication bus.
+    DltNwTraceCan,
+    ///FlexRay communication bus.
+    DltNwTraceFlexray,
+    ///Most communication bus.
+    DltNwTraceMost,
+    ///Ethernet communication bus.
+    DltNwTraceEthernet,
+    ///SOME/IP communication.
+    DltNwTraceSomeIp,
+    ///User defined settings.
+    DltNwUserDefined(u8),
+}
+
+impl MtinDltNetworkMessage {
+    ///Attempts to convert a raw mtin value to the enum.
+    ///Returns `Option::None` for reserved values.
+    fn from_u8(value: u8) -> Option<Mtin> {
+        match value {
+            0x1 => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwTraceIpc)),
+            0x2 => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwTraceCan)),
+            0x3 => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwTraceFlexray)),
+            0x4 => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwTraceMost)),
+            0x5 => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwTraceEthernet)),
+            0x6 => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwTraceSomeIp)),
+            //User defined
+            0x7 ..= 0xF => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwUserDefined(value))),
+            //Mtin is only 4 bit
+            _ => None
+        }
+    }
+
+    fn to_u8(&self) -> u8 {
+        match self {
+            MtinDltNetworkMessage::DltNwTraceIpc => 0x1,
+            MtinDltNetworkMessage::DltNwTraceCan => 0x2,
+            MtinDltNetworkMessage::DltNwTraceFlexray => 0x3,
+            MtinDltNetworkMessage::DltNwTraceMost => 0x4,
+            MtinDltNetworkMessage::DltNwTraceEthernet => 0x5,
+            MtinDltNetworkMessage::DltNwTraceSomeIp => 0x6,
+            //User defined
+            MtinDltNetworkMessage::DltNwUserDefined(value) => *value,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum MtinDltControlMessage {
+    ///Request control message.
+    DltControlRequest = 0x1,
+    ///Respond control message.
+    DltControlResponse = 0x2
+}
+
+impl MtinDltControlMessage {
+    ///Attempts to convert a raw mtin value to the enum.
+    ///Returns `Option::None` for reserved values.
+    fn from_u8(value: u8) -> Option<Mtin> {
+        match value {
+            0x1 => Some(Mtin::DltControlMessage(MtinDltControlMessage::DltControlRequest)),
+            0x2 => Some(Mtin::DltControlMessage(MtinDltControlMessage::DltControlResponse)),
+            //Reserved values
+            _ => None,
+        }
+    }
+}
+
+///Message type info field
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum Mtin {
```

Comment:
This enum could be flattened. Not sure what's more readable here. I think the current way might be easier to match against.

Reply:
I am also not 100% sure what the best choice here is. Maybe `Mtin` could be renamed to `Msin`or `Mstp` and the the separate `Mstp` enum could be removed, as it is redundant. The tradeoff would be, that you would always have to deserialized both variables, even if you are only interested in the first one.

I would leave it like this for now.